core/frontend/src/Marker.ts

```diff
@@ -307,13 +307,25 @@ export class Marker implements CanvasDecoration {
  */
 export class Cluster<T extends Marker> {
   public readonly markers: T[];
-  public readonly rect: ViewRect;
   public clusterMarker?: Marker;
 
   public constructor(markers: T[]) {
-    this.rect = markers[0].rect;
     this.markers = markers;
   }
+
+  public get position() {
+    return this.markers[0].position;
+  }
+  /**
+   * Gets the location for the cluster
+   * @returns The average of the cluster markers worldLocation.
+   */
+  public getClusterLocation() {
+    const location = Point3d.createZero();
+    this.markers.forEach((marker) => location.addInPlace(marker.worldLocation));
+    location.scaleInPlace(1 / this.markers.length);
```

Comment:
divide by zero check?

Reply:
should probably be an assert on the constructor that the array isn't empty

Reply:
Added assert to constructor.

Reply:
`markers` can be modified after constructor. Dividing by zero will potentially terminate the entire app with uncaught exception, and is trivially avoidable by adding a check here.

Reply:
Good point, I added code to guard against it.